bolt-modules/out/lib/puppet/functions/out/verbose.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+# Output a message for the user when running with --verbose
+#
+# This will print a message to stdout when using the human output format,
+# and print to stderr when using the json output format
+#
+# > **Note:** Not available in apply block
+Puppet::Functions.create_function(:'out::verbose') do
+  # Output a message when run with --verbose.
+  # @param message The message to output when run with --verbose.
+  # @example Print a message
+  #   out::verbose('Something went wrong')
+  dispatch :output_message do
+    param 'Any', :message
+    return_type 'Undef'
+  end
+
+  def output_message(message)
+    unless Puppet[:tasks]
+      raise Puppet::ParseErrorWithIssue
+        .from_issue_and_stack(Bolt::PAL::Issues::PLAN_OPERATION_NOT_SUPPORTED_WHEN_COMPILING, action: 'out::verbose')
+    end
+
+    executor = Puppet.lookup(:bolt_executor)
+    # Send Analytics Report
+    executor.report_function_call(self.class.name)
+
+    executor.publish_event(type: :verbose, message: stringify(message))
```

Comment:
`stringify()` is also defined in the `out::message` function. Since we now use this method in multiple locations, it might make sense to split it out into the `Bolt::Util` module, which includes reusable utility methods.

I did something similar to that in #2908 which you can see [here](https://github.com/puppetlabs/bolt/blob/90aeffe471daa810be906a1163f8d5ab1c24c220/lib/bolt/util/format.rb). Then you would just require the module in this file and call the function like `Bolt::Util.stringify(message)`.

Reply:
That makes a lot of sense. Is the easiest way to do that (since your PR isn't merged yet), to just copy the whole format file into `Util` directory on my PR?

Reply:
That's typically what I do. If my PR gets merged first then you can rebase off `main` to pull the changes in or vice versa. :)

Reply:
I'm also not sure if we want to just put that in `Bolt::Util` or start breaking out into smaller modules. 🤷 Figured we'd sort that out in review.

Reply:
@lucywyman thoughts?

Reply:
Related thread: https://github.com/puppetlabs/bolt/pull/2908#discussion_r662571553